src/plugins/identity/identity.test.js

```diff
@@ -1,33 +1,41 @@
 // @flow
 
 import {NodeAddress} from "../../core/graph";
-import {identityNode} from "./identity";
+import {identityAddress, identityNode} from "./identity";
 
 describe("src/plugins/identity/identity", () => {
-  describe("identityNode", () => {
+  describe("identityNode & identityAddress", () => {
     it("works as expected for valid identity", () => {
       const identity = {username: "foo", aliases: ["github/foo"]};
       const n = identityNode(identity);
       expect(n.address).toEqual(
         NodeAddress.fromParts(["sourcecred", "identity", "foo"])
       );
+      expect(identityAddress(identity.username)).toEqual(n.address);
       expect(n.timestampMs).toEqual(null);
       expect(n.description).toEqual("@foo");
     });
     it("errors for an empty username", () => {
       const identity = {username: "", aliases: ["github/foo"]};
       expect(() => identityNode(identity)).toThrowError("Invalid username");
+      expect(() => identityAddress(identity.username)).toThrowError(
+        "Invalid username"
+      );
     });
     it("errors for a bad username", () => {
       const identity = {username: "$foo$bar", aliases: ["github/foo"]};
       expect(() => identityNode(identity)).toThrowError("Invalid username");
+      expect(() => identityAddress(identity.username)).toThrowError(
+        "Invalid username"
+      );
```

Comment:
Nit: have some preference to split up the tests for `validateUsername`, `identityNode` and `identityAddress`.

The pattern I usually go for is to:
- Test `validateUsername` with a number of good and bad example inputs.
- Test `identityAddress` does username validation at all, testing one bad input throws.
- Test `identityNode` does username validation at all, testing one bad input throws.

Reply:
Exported `validateUsername` and added explicit tests.
Left the existing tests as is--I'm ok with having a bit of redundant testing.